### PR TITLE
WINDUPRULE-405: eap 7.3 deprecated ha singleton api rules

### PR DESCRIPTION
**Pull request by emmartins**
Issue: https://issues.jboss.org/browse/WINDUPRULE-405

### CODE REVIEW

**Comment by emmartins** [comment]
please do not merge, <link/> to be fixed once EAP 7.3 is released

File: rules-reviewed/eap7/eap72/deprecated-singletonpolicy.rhamt.xml [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+        <!-- https://issues.jboss.org/browse/WINDUPRULE-405 -->
+        <rule id="deprecated-singletonpolicy-00001">
+            <when>
+                <javaclass references="org.wildfly.clustering.singleton.SingletonPolicy" >
+                    <location>IMPORT</location>
+                </javaclass>
+            </when>
+            <perform>
+                <hint title="Deprecated HA Singleton API" effort="1" category-id="potential">
+                    <message>The application imports class 'org.wildfly.clustering.singleton.SingletonPolicy', from deprecated HA Singleton API.</message>
```
Comment (mrizzi) [inline rules-reviewed/eap7/eap72/deprecated-singletonpolicy.rhamt.xml]:
```suggestion
                    <message>The application imports class `org.wildfly.clustering.singleton.SingletonPolicy`, from deprecated HA Singleton API.</message>
```

File: rules-reviewed/eap7/eap72/deprecated-singletonpolicy.rhamt.xml [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+                    <link title="FIXME" href="FIXME"/>
```
Comment (mrizzi) [inline rules-reviewed/eap7/eap72/deprecated-singletonpolicy.rhamt.xml]: @emmartins i feel like this rule should be ready for review, have you got the link's URL?

File: rules-reviewed/eap7/eap72/deprecated-singletonpolicy.rhamt.xml (hunk 1 above)
Comment (emmartins) [reply rules-reviewed/eap7/eap72/deprecated-singletonpolicy.rhamt.xml]: done

File: rules-reviewed/eap7/eap72/deprecated-singletonpolicy.rhamt.xml (hunk 2 above)
Comment (emmartins) [reply rules-reviewed/eap7/eap72/deprecated-singletonpolicy.rhamt.xml]: probably, the content is in EAP Developer Guide, will check with docs team what is the link

Comment (emmartins) [reply rules-reviewed/eap7/eap72/deprecated-singletonpolicy.rhamt.xml]: link added

Comment (mrizzi) [reply rules-reviewed/eap7/eap72/deprecated-singletonpolicy.rhamt.xml]: It works adding the staged option :+1:

**Comment by mrizzi** [comment]
retest

**Comment by mrizzi** [comment]
It has been decided to merge this PR even if the link test fails because the documentation link has been manually tested.

**Pull request merged**